src/Microsoft.DocAsCode.Build.Engine/TemplateRenderers/LiquidTemplateRenderer.cs

```diff
@@ -7,32 +7,57 @@ namespace Microsoft.DocAsCode.Build.Engine
     using System.Collections.Concurrent;
     using System.Collections.Generic;
     using System.Collections.Immutable;
+    using System.Linq;
 
     using Microsoft.DocAsCode.Common;
 
     internal class LiquidTemplateRenderer : ITemplateRenderer
     {
-        private static object _locker = new object();
-        private readonly DotLiquid.Template _template = null;
-        public static LiquidTemplateRenderer Create(ResourceCollection resourceProvider, string template)
+        private static readonly object _locker = new object();
+
+        private readonly DotLiquid.Template _template;
+
+        public static LiquidTemplateRenderer Create(ResourceCollection resourceProvider, string template, string templateName)
         {
-            if (template == null) throw new ArgumentNullException(nameof(template));
+            if (template == null)
+            {
+                throw new ArgumentNullException(nameof(template));
+            }
+
+            if (templateName == null)
+            {
+                throw new ArgumentNullException(nameof(templateName));
+            }
+
+            // Guarantee that each time returns a new renderer
+            // As Dependency is a globally shared object, allow one entry at a time
             lock (_locker)
             {
                 DotLiquid.Template.RegisterTag<Dependency>("ref");
                 Dependency.PopDependencies();
                 var liquidTemplate = DotLiquid.Template.Parse(template);
```

Comment:
_locker is a static lock, why there is another lock?

---
In reply to: [109279261](https://github.com/dotnet/docfx/pull/1513#discussion_r109279261) [](ancestors = 109279261)